### PR TITLE
Sre 226 split invoice and transaction endpoints

### PR DESCRIPTION
**Pull request by PSchierz**
I extracted all methods that are used by several endpoints to the utils. I dont think that the structure is quite optimal but I think it suffices for globalSearch (which is hopefully a story that ends at some point).

### CODE REVIEW

File: globalsearch/utils/check_utils.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import flask
+
+
+def check_dict_has_none_values(dictionary):
```
Comment (jannisrelakis) [inline globalsearch/utils/check_utils.py:5]: Could be simplified as `return None in dictionary.values()`

File: globalsearch/utils/check_utils.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    has_none_values = check_dict_has_none_values(entity)
+
+    if has_none_values:
+        print(f"Update with null values was rejected")
+        flask.abort(flask.make_response(flask.jsonify(error={
+            "code": "NullValuesNotAllowed",
+            "message": "The request contained null values which are not allowed."
+        }), 400))
+
+    return
```
Comment (jannisrelakis) [inline globalsearch/utils/check_utils.py]: Sometimes you explicitly return nothing, sometimes you don't. See line 19. Why?

File: globalsearch/utils/check_utils.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        print(f"Rejected request with invalid API key")
+        flask.abort(flask.make_response(flask.jsonify(error={
+            "code": "AuthenticationError",
+            "message": "Invalid API key."
+        }), 401))
+    return
+
+
+def check_if_id_is_present(entity_id):
+    if entity_id is None or entity_id == "":
```
Comment (jannisrelakis) [inline globalsearch/utils/check_utils.py]: Could be simplified as `if not entity_id`.

File: globalsearch/utils/check_utils.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        print(f"update without specified id failed")
```
Comment (jannisrelakis) [inline globalsearch/utils/check_utils.py]: Although we haven't agreed on this, I think log messages should start with a capital letter and end with a full-stop.

**Review by jannisrelakis — APPROVED** [review]
LGTM. Just some cosmetic changes.

File: globalsearch/utils/optimistic_locking_utils.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import flask
+
+
+def respond_with_optimistic_locking_error(entity, entity_name, max_retries_on_version_conflict):
+    print(f"Failed to create/update/delete {entity_name}: {entity} after {max_retries_on_version_conflict} retries.")
+    flask.abort(flask.make_response(flask.jsonify(error={
+        "code": "VersionConflict",
+        "message": f"Service failed to consistently update the {entity_name} due to 3 failed optimistic locking trials"
```
Comment (jannisrelakis) [inline globalsearch/utils/optimistic_locking_utils.py]: You hardcoded the 3 in the message.

File: globalsearch/utils/check_utils.py (hunk 1 above)
Comment (PSchierz) [reply globalsearch/utils/check_utils.py:5]: Yeah thanks! That is much better!

File: globalsearch/utils/check_utils.py (hunk 2 above)
Comment (PSchierz) [reply globalsearch/utils/check_utils.py]: I think PyCharm generated the return when I extracted the methods. I got rid of them. Thanks for the note!

File: globalsearch/utils/check_utils.py (hunk 4 above)
Comment (PSchierz) [reply globalsearch/utils/check_utils.py]: Yes thanks I changed it.

File: globalsearch/utils/optimistic_locking_utils.py (hunk 5 above)
Comment (PSchierz) [reply globalsearch/utils/optimistic_locking_utils.py]: Thank you that you spotted that .... I changed it.

File: globalsearch/utils/check_utils.py (hunk 3 above)
Comment (PSchierz) [reply globalsearch/utils/check_utils.py]: Thanks again. I changed that as well.

**Pull request merged**